odl/tomo/operators/ray_trafo.py

```diff
@@ -273,299 +158,217 @@ def __init__(self, reco_space, geometry, variant, **kwargs):
-            Implementation back-end for the transform. Supported back-ends:
-
-            - ``'astra_cuda'``: ASTRA toolbox, using CUDA, 2D or 3D
-            - ``'astra_cpu'``: ASTRA toolbox using CPU, only 2D
-            - ``'skimage'``: scikit-image, only 2D parallel with square
-              reconstruction space.
-
-            For the default ``None``, the fastest available back-end is
-            used, tried in the above order.
+        bool : use_cache
+            If ``True`` returns the cached implementation backend, if it
+            was generated in a previous call (or given with ``__init__``).
+            If ``False`` a new instance of the backend will be generated,
+            freeing up GPU memory and RAM used by the backend.
+        """
 
-        domain : `DiscreteLp`, optional
-            Discretized projection (sinogram) space, the domain of the
-            backprojection operator.
-            Default: Inferred from parameters.
-        use_cache : bool, optional
-            If ``True``, data is cached. This gives a significant speed-up
-            at the expense of a notable memory overhead, both on the GPU
-            and on the CPU, since a full volume and a projection dataset
-            are stored. That may be prohibitive in 3D.
-            Default: True
-        kwargs
-            Further keyword arguments passed to the projector backend.
+        # Use impl creation (__cached_impl) when `use_cache` is True
+        if not use_cache or self.__cached_impl is None:
+            # Lazily (re)instantiate the backend
+            self.__cached_impl = self._impl_type(
+                self.geometry,
+                vol_space=self.domain,
+                proj_space=self.range)
 
-        Notes
-        -----
-        The ASTRA backend is faster if data are given with
-        ``dtype='float32'`` and storage order 'C'. Otherwise copies will be
-        needed.
+        return self.__cached_impl
 
-        See Also
-        --------
-        astra_cpu_back_projector
-        AstraCudaBackProjectorImpl
-        skimage_radon_back_projector
-        """
-        domain = kwargs.pop('domain', None)
-        super(RayBackProjection, self).__init__(
-            reco_space=range, proj_space=domain, geometry=geometry,
-            variant='backward', **kwargs)
+    def _call(self, x, out=None, **kwargs):
+        """Forward projection
 
-    def _call_real(self, x_real, out_real, **kwargs):
-        """Real-space back-projection for the current set-up.
+        Parameters
+        ----------
+        x : DiscreteLpElement
+            A volume. Must be an element from `RayTransform.domain`.
+        out : `RayTransform.range` element, optional
+            A DiscreteLpElement from `RayTransform.range`, to which the result
+            of the operator evaluation is written.
+        **kwargs
+            Arbitrary keyword arguments, passed on to the implementation
+            backend.
```

Comment:
```suggestion
            Extra keyword arguments, passed on to the implementation
            backend.
```
"Arbitrary" goes a bit too far IMO

Reply:
Agreed. Changed at two places.